scripts/check_ci_test_suites_to_run.py

```diff
@@ -0,0 +1,691 @@
+        root_file: str. The root file.
+        test_suites_to_module_mapping: dict. The test suites to module mapping
+            for the test type.
+        test_suites: list(dict). The list of test suites.
+
+    Returns:
+        list(dict). The test suites affected by the root file.
+    """
+    test_suites_affected: List[GenericTestSuiteDict] = []
+    for test_suite in test_suites:
+        if test_suite['name'] not in test_suites_to_module_mapping:
+            test_suites_affected.append(test_suite)
+    for (
+        test_suite_name,
+        test_suite_modules
+    ) in test_suites_to_module_mapping.items():
+        test_suite_by_name = get_test_suite_by_name_from_list(
+            test_suites, test_suite_name
+        )
+        if (
+            test_suite_by_name is not None and (
+                root_file in test_suite_modules or
+                root_file == test_suite_by_name['module']
+            )
+        ):
+            test_suites_affected.append(test_suite_by_name)
+
+    return extend_test_suites_without_duplicates(
+        [],
+        test_suites_affected
+    )
+
+
+def get_affected_lighthouse_pages(
+    modified_root_files: Set[str]
+) -> List[LighthousePageDict]:
+    """Gets the affected Lighthouse pages by a list of modified root files.
+
+    Args:
+        modified_root_files: set(str). The set of modified root files.
+
+    Returns:
+        list(dict). The affected Lighthouse pages sorted by name.
+    """
+    lighthouse_pages = get_lighthouse_pages_from_config()
+    affected_lighthouse_pages: List[LighthousePageDict] = []
+    for modified_root_file in modified_root_files:
+        for lighthouse_page in lighthouse_pages:
+            if modified_root_file == lighthouse_page['page_module']:
+                affected_lighthouse_pages.append(lighthouse_page)
+
+    return sorted(affected_lighthouse_pages, key=lambda x: x['name'])
+
+
+def get_ci_test_suites_to_run(
+    modified_files: List[str],
+    root_files_mapping: dict[str, List[str]]
+) -> CITestSuitesToRunDict | None:
+    """Gets the test suites to run in the CI.
+
+    Args:
+        modified_files: list(str). The list of modified files.
+        root_files_mapping: dict. The root files mapping.
+
+    Returns:
+        dict | None. The test suites to run in the CI or None if all test
+        suites should be run.
+    """
+    modified_root_files: Set[str] = set()
+    for file in modified_files:
+        # If the file is not in the root files mapping, then all test suites
+        # should be run. This is the case for files that are not scraped by
+        # the root files mapping script (e.g. files like package.json).
+        if file not in root_files_mapping:
+            return None
+        modified_root_files.update(root_files_mapping[file])
+
+    root_files_config = get_root_files_config()
+    for modified_root_file in modified_root_files:
+        if modified_root_file in root_files_config['run_all_tests_root_files']:
```

Comment:
Can this be simplified into a set operation?

Reply:
Done, Thanks.